src/sql/workbench/services/objectExplorer/browser/serverTreeRenderer.ts

```diff
@@ -171,16 +171,14 @@ export class ServerTreeRenderer implements IRenderer {
 		}
 
 		let iconId = this._connectionManagementService.getConnectionIconId(connection.id);
-		if (!iconId) { return undefined; }
-
 		let providerProperties = this._connectionManagementService.getProviderProperties(connection.providerName);
 		if (!providerProperties) { return undefined; }
 
 		let iconPath: IconPath | undefined = undefined;
-		let pathConfig: URI | IconPath | { id: string, path: IconPath }[] | undefined = providerProperties.iconPath;
+		let pathConfig: URI | IconPath | { id: string, path: IconPath, default?: boolean }[] | undefined = providerProperties.iconPath;
 		if (Array.isArray(pathConfig)) {
 			for (const e of pathConfig) {
-				if (!e.id || e.id === iconId) {
+				if (!e.id || e.id === iconId || e.default) {
```

Comment:
Actually the logic here needs to be 

`if (!e.id || e.id === iconId || (!iconId && e.default)) {`

otherwise if the default icon is before the actual icon we'll use the default instead.